src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsRulesSpec.groovy

```diff
@@ -988,4 +1000,139 @@ class PriceFloorsRulesSpec extends PriceFloorsBaseSpec {
         where:
         enforcePbs << [true, null]
     }
+
+    def "PBS should use correct rule for each bidder when bidder is define with valid bid floor value"() {
+        given: "PBS config with openX bidder"
+        def floorsPbsService = pbsServiceFactory.getService(
+                FLOORS_CONFIG + GENERIC_ALIAS_CONFIG +
+                        ["adapters.openx.enabled" : "true",
+                         "adapters.openx.endpoint": "$networkServiceContainer.rootUri/auction".toString()])
+
+        and: "BidRequest with generic and openx bidder"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            addImp(Imp.defaultImpression.tap {
+                ext.prebid.bidder.generic = null
+                ext.prebid.bidder.openx = Openx.defaultOpenx
+            })
+        }
+
+        and: "Account with enabled fetch, fetch.url in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.accountId)
+        accountDao.save(account)
+
+        and: "Set Floors Provider response"
+        def floorsResponse = PriceFloorData.priceFloorData.tap {
+            modelGroups[0].schema = new PriceFloorSchema(fields: [BIDDER])
+            modelGroups[0].values = [(new Rule(bidder: GENERIC).rule): genericBidFloorRuleValue,
+                                     (new Rule(bidder: OPENX).rule)  : openxBidFloorRuleValue]
+        }
+        floorsProvider.setResponse(bidRequest.accountId, floorsResponse)
+
+        and: "PBS fetch rules from floors provider"
+        cacheFloorsProviderRules(floorsPbsService, bidRequest)
+
+        when: "PBS processes auction request"
+        def response = floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request bidFloor should correspond to appropriate rule"
+        def bidderRequest = getRequest(response)
+        assert bidderRequest.size() == 2
+        assert bidderRequest[GENERIC.value].first.imp[0].bidFloor == genericBidFloorRuleValue
+        assert bidderRequest[OPENX.value].first.imp[0].bidFloor == openxBidFloorRuleValue
+
+        and: "Bidder request should contain proper bid floor value"
+        def bidderRequests = bidder.getBidderRequests(bidRequest.id)
+        assert bidderRequests.imp.bidFloor.toSet().flatten()
+                .any { bidFloor -> bidFloor == genericBidFloorRuleValue || bidFloor == openxBidFloorRuleValue }
```

Comment:
This check is very strange. This will pass even if Generic will get OpenX value and vice-versa.